osu.Game/Beatmaps/Formats/BeatmapDecoder.cs

```diff
@@ -25,9 +26,10 @@ public static BeatmapDecoder GetDecoder(StreamReader stream)
             do { line = stream.ReadLine()?.Trim(); }
             while (line != null && line.Length == 0);
 
-            if (line == null || !decoders.ContainsKey(line))
+            string decoderKey = decoders.Keys.FirstOrDefault(k => line?.Contains(k) ?? false);

```

Comment:
probably better off using `decoders.TryGetValue()`. should be more efficient and make the code a bit cleaner.

Reply:
I'm going to do that once I'm home.
Question time: This may find a wrong version. I noticed v1 doesn't exist but in the future once versions go up to 30+ it could be random which decoder is found between v3, v30, v31, etc.
Is it desired for it to be more lenient? I could think of something with regex and try greedy matching. That may be more precise.

Reply:
If you're using `TryGetValue` that will not be an issue.

Reply:
I'm not sure if we talk about the same thing. Just to make sure, please bear with me.

The original code got a decoder by exact value.
`!decoders.ContainsKey(line)`

I'm trying to find a decoder that may be occuring in the line so accidental characters (like in the example in the initial message) don't cause it to fail.
`decoders.Keys.FirstOrDefault(k => line?.Contains(k) ?? false)`

`TryGetValue` would go back to the start as it would only look for an exact match. That confused me so I was asking if it is okay to try to be more lenient. If it's not wanted anyways this PR would be unnecessary in the first place.